Lib/idlelib/run.py

```diff
@@ -250,30 +250,98 @@ def print_exception():
     sys.last_exc = val
     seen = set()
 
-    def print_exc(typ, exc, tb):
+    def print_exc(typ, exc, tb, prefix=""):
         seen.add(id(exc))
         context = exc.__context__
         cause = exc.__cause__
+        exclude = ("run.py", "rpc.py", "threading.py", "queue.py",
+                   "debugger_r.py", "bdb.py")
         if cause is not None and id(cause) not in seen:
-            print_exc(type(cause), cause, cause.__traceback__)
-            print("\nThe above exception was the direct cause "
-                  "of the following exception:\n", file=efile)
+            print_exc(type(cause), cause, cause.__traceback__, prefix)
+            if prefix:
+                print(f"{prefix}|\n{prefix}| The above exception was the direct cause "
+                      f"of the following exception:\n{prefix}|", file=efile)
+            else:
+                print("\nThe above exception was the direct cause "
+                      "of the following exception:\n", file=efile)
         elif (context is not None and
               not exc.__suppress_context__ and
               id(context) not in seen):
-            print_exc(type(context), context, context.__traceback__)
-            print("\nDuring handling of the above exception, "
-                  "another exception occurred:\n", file=efile)
-        if tb:
-            tbe = traceback.extract_tb(tb)
-            print('Traceback (most recent call last):', file=efile)
-            exclude = ("run.py", "rpc.py", "threading.py", "queue.py",
-                       "debugger_r.py", "bdb.py")
-            cleanup_traceback(tbe, exclude)
-            traceback.print_list(tbe, file=efile)
-        lines = get_message_lines(typ, exc, tb)
-        for line in lines:
-            print(line, end='', file=efile)
+            print_exc(type(context), context, context.__traceback__, prefix)
+            if prefix:
+                print(f"{prefix}|\n{prefix}| During handling of the above exception, "
+                      f"another exception occurred:\n{prefix}|", file=efile)
+            else:
+                print("\nDuring handling of the above exception, "
+                      "another exception occurred:\n", file=efile)
+        if isinstance(exc, BaseExceptionGroup):
+            if tb:
+                if not prefix:
+                    print("  + Exception Group Traceback (most recent call last):", file=efile)
+                else:
+                    print(f"{prefix}| Exception Group Traceback (most recent call last):", file=efile)
+                tbe = traceback.extract_tb(tb)
+                cleanup_traceback(tbe, exclude)
+                for line in traceback.format_list(tbe):
+                    for subline in line.rstrip().splitlines():
+                        if not prefix:
+                            print(f"  | {subline}", file=efile)
+                        else:
+                            print(f"{prefix}| {subline}", file=efile)
```

Comment:
```suggestion
                prefix2 = prefix or "  "
                for line in traceback.format_list(tbe):
                    for subline in line.rstrip().splitlines():
                        print(f"{prefix2}| {subline}", file=efile)
```

You can store "prefix2" (or whatever other name you choose) at the beginning of the function. Then you can reuse below when you print the message lines as well.
That way, you avoid having `if not prefix` at every loop iteration.